### ISSUE TITLE
Show more/all history for 'My Trades' in the DEX

### ISSUE THREAD

**Issue by svk31** [bitshares/bitshares-ui#68]
_From @steempower on January 13, 2017 5:20_

The 'My trades' box within the DEX should provide a longer history of trades by default and include a button to show all trades which would fetch the entire history of the users trades for that market pair

_Copied from original issue: cryptonomex/graphene-ui#895_

**Comment by wmbutler** [comment]
This needs an infinite scrolling strategy.

**Comment by happyconcepts** [comment]
When exporting the history as csv, the filename defaults to using zero-based month which may be confusing to end users. 

Example: btshist-2017-6-22.csv is generated for a history exported on July 22.

**Comment by abitmore** [comment]
> When exporting the history as csv, the filename defaults to using zero-based month which may be confusing to end users.
>
> Example: btshist-2017-6-22.csv is generated for a history exported on July 22.

Best report it as a separate issue. Thanks.

**Comment by Ashaman-** [comment]
Is it possible to have the wallet save trade history locally? And if there are gaps due to a wallet being offline when a bunch of orders execute, maybe it can use `get_relative_account_history` to fill them in?

**Comment by wmbutler** [comment]
@Ashaman- Your question is unrelated to the issue. I do not think the wallet should store history. The wallet should query the API to obtain the history.

**Comment by wmbutler** [comment]
As I think about this, paging would be the easiest to implement.

**Comment by abitmore** [comment]
> As I think about this, paging would be the easiest to implement.

It may require API support/changes. but I'm not sure. Please open an issue in the bitshares-core repository. Or ask in the dev channel first.

**Comment by wmbutler** [comment]
Mentioned this in the dev channel.

**Comment by wmbutler** [comment]
**get_relative_account_history**(account_id_type account, uint32_t stop = 0, unsigned limit = 100, uint32_t start = 0) const

Get operations relevant to the specified account referenced by an event numbering specific to the account. The current number of operations for the account can be found in the account statistics (or use 0 for start).
### Return
A list of operations performed by account, ordered from most recent to oldest.
### Parameters
- account: The account whose history should be queried
- stop: Sequence number of earliest operation. 0 is default and will query ‘limit’ number of operations.
- limit: Maximum number of operations to retrieve (must not exceed 100)
- start: Sequence number of the most recent operation to retrieve. 0 is default, which will start querying from the most recent operation.

**Comment by wmbutler** [comment]
In order to fully satisfy this requirement, we need to be able to **optionally** request:

- type of operation
- date range
- trading pair

in addition to the features of **get_relative_account_history**

**Comment by abitmore** [comment]
Just a note here, there is a `get_account_history` API with a type filter, but the implementation is inefficient and hard to improve as of writing, so I won't recommend that. Yes, we can improve the `get_relative_account_history` API, but perhaps need to create a new API for backward compatibility.

**Comment by wmbutler** [comment]
@oxarbitrage just touching base on the api progress for this?

**Comment by oxarbitrage** [comment]
yes, sorry the delay @wmbutler. i had been working very hard over the last 2 weeks to get this done the right way, this is affecting performance and memory usage the less possible while returning the exact results inside the requested time range. it haves some technical difficulties because of how the implementation is designed, i have a solution, actually i already presented several, this last one is the most close to optimal we can think of. solution will also be of use for other issues we have in bitshares-core. i am getting guidance by @pmconrad and i know i told this several times but it should be ready very soon, this week i hope. i can say i am working non stop to make this pull request happened and move on to something else.

**Comment by wmbutler** [comment]
Not a problem. I just want to know if I should delay it. I'll move it to 171015 for now.

**Comment by wmbutler** [comment]
@oxarbitrage this issue specifically is requesting for a given trading pair. Did you build in the ability to filter to a trading pair as well? I know this is going to help for the overall trading history but couldn't remember the answer to this question.

**Comment by oxarbitrage** [comment]
Hi @wmbutler sorry for the late response, mention was while i was travelling and i didn't saw it until now. as you probably know we are working in a global solution to browse the account history in all ways needed included filtering by the assets, operation type or anything. 
To do this we are moving the account data history to disk, reducing the ram needed by a full node too. Ill recommend you to move all the account_history related issues to 171101 milestone.

**Comment by wmbutler** [comment]
@oxarbitrage checking in on this again. Will this be ready for the 171101 release?

**Comment by wmbutler** [comment]
@oxarbitrage will I need to push this to 171115?

**Comment by svk31** [comment]
Yes, there's no timeline right now for when this will be available, and once the actual code is available we'll still need the api nodes to upgrade to it..

**Comment by testlump** [comment]
Is there an update on the progress of this issue?

**Comment by rokii666** [comment]
Hello! Need to see all my trade history. Also if possible to group by my orders. Say I place order for 1000 bitUSD and it buys for 2 or 3 deals and they are shown as separate. But because they are from my one order I want to group them somehow so I can see my total order for 1000 bitUSD

**Comment by wmbutler** [comment]
I've placed this in 180215 Sprint. It's all part of the ElasticSearch integration. Really need @oxarbitrage and @svk31 to pair on this to knock it out.

**Comment by oxarbitrage** [comment]
sure, here is what needs to be done.

there is by now 1 public elastic search wrapper api node:
http://185.208.208.184:5000/get_account_history?account_id=1.2.282&from=100&size=10&sort_by=block_data.block_time

but ideally there is going to be some more, so think on a pool of address:port for and pick by latency/whatever.

query the database for account by using the samples here:
https://github.com/oxarbitrage/bitshares-es-wrapper#available-calls

if more options are needed let me know.

**Comment by sschiessl-bcp** [comment]
We are deploying a elastic search node with the infrastructure worker, details will be in our final report under 
https://bitsharestalk.org/index.php?topic=25291.0

That could then be used for this feature

**Comment by sschiessl-bcp** [comment]
Anyone working on this?

**Comment by wmbutler** [comment]
No, but it's needed.

**Comment by kohako** [comment]
While waiting for this update is there a way to view the market trades for a pair somewhere else ?

**Comment by startailcoon** [comment]
http://open-explorer.io/ is a good start to look at

**Comment by startailcoon** [comment]
ES Wrapper call implemented as `get_account_history`. https://eswrapper.bitshares.eu/apidocs/#!/wrapper/get_account_history

**Comment by clockworkgr** [comment]
Is anyone working on this? With the increased chain size since OP creation this is impossible with the standard API.

Will either have to be done via ES calls or not at all.

Do we have an epic about integrating ES data ?

**Comment by sschiessl-bcp** [comment]
Yes there is one, we have been mostly waiting to ensure that ES backend has matured and doesn't change interfaces / indices anymore. I suppose that could now be re-evaluated.

For the reference UI we would need reliable ES backends though, and right now the only one is the infrastructure worker (which is not active atm), plus 2 from the community with no guarantee of service. So in that state it is not feasible to work on heavy ES integration.

The sole ES integration right now is account history export.
